### PR TITLE
 aii-server: add opennebula rpm

### PR DESCRIPTION
**Pull request by kwaegema**
Depends on https://github.com/quattor/aii/pull/156

### CODE REVIEW

**Comment by jrha** [comment]
@kwaegema rebase plz :recycle:

**Comment by kwaegema** [comment]
rebased. This still fails, but I don't know why.. Something with /undef is undefined ..?

File: quattor/aii/opennebula/rpms.pan [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# 
+# #
+# aii-opennebula, 16.2.0-rc4-SNAPSHOT, rc4_SNAPSHOT20160219150415, 2016-02-19T15:04:15Z
+#
+
+# Template adding aii-opennebula rpm to the configuration
+
+unique template quattor/aii/opennebula/rpms;
+
+"/software/packages"=pkg_repl("aii-opennebula","16.2.0-rc4_SNAPSHOT20160219150415","noarch");
```
Comment (kwaegema) [inline quattor/aii/opennebula/rpms.pan]: ok, these templates should come together with the other aii templates, for a specific release I guess. @jrha  don't know why they are not here yet, aii-opennebula is in master?

Comment (stdweird) [inline quattor/aii/opennebula/rpms.pan]: @kwaegema i think aii-opennebula is not part of the aii main pom.xml

Comment (kwaegema) [inline quattor/aii/opennebula/rpms.pan]: it is in master. mvn_pack gives:
[INFO] Reactor Summary:
[INFO] 
[INFO] AII server part .................................... SUCCESS [  2.987 s]
[INFO] AII DHCP config generator .......................... SUCCESS [  1.007 s]
[INFO] AII Kickstart generator ............................ SUCCESS [ 32.427 s]
[INFO] AII PXE-linux generator ............................ SUCCESS [ 24.586 s]
[INFO] Opennebula hook for AII ............................ SUCCESS [ 16.325 s]
[INFO] AII ................................................ SUCCESS [  0.077 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS

**Comment by jrha** [comment]
test this please

File: quattor/aii/opennebula/default.pan [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+variable MAC_PREFIX ?= '02:00';
+
+"/system/aii/hooks/configure/" = {
+    append(nlist(
+        'module', OPENNEBULA_AII_MODULE_NAME,
+
+        "image", OPENNEBULA_AII_FORCE,
+        "template", OPENNEBULA_AII_FORCE,
+        ));
```
Comment (jrha) [inline quattor/aii/opennebula/default.pan]: Indentation here is weird, also please use `dict` throughout this code.

This also seems overly complicated considering you are appending a single item.

How about this?
```pan
'/system/aii/hooks/configure/' = append(dict(
    'module', OPENNEBULA_AII_MODULE_NAME,
    'image', OPENNEBULA_AII_FORCE,
    'template', OPENNEBULA_AII_FORCE,
));
```

**Comment by kwaegema** [comment]
I think the templates of this pr should come together with the aii templates, version locked on release, don't know why aii-opennebula is not included, it is in the pom.xml of aii

**Comment by jouvin** [comment]
It would be good to update the title (I guess that it is `aii-opennebula` RPM) and have a short description summarizing the motivation...

**Comment by jouvin** [comment]
And it would be good to fix the problem preventing the tests to work...

**Comment by jrha** [comment]
@kwaegema can you explain what this is for please?

File: quattor/aii/opennebula/default.pan (hunk 2 above)
Comment (kwaegema) [inline quattor/aii/opennebula/default.pan]: this is copied from aii-opennebula in aii repo

**Comment by kwaegema** [comment]
Actually, I don't know exactly what should happen here. I saw that the aii/opennebula templates from aii were not in the template library core, where the other aii modules have there pan files here. I made a PR to copy them here, but in the process I've had the impression that's not the way this gets here, because it is build for a specific release. @jrha Is that something you know? Then the remaining question is: why isn't opennebula also in the release?

**Comment by jrha** [comment]
Oh, I finally understand!

**Pull request closed without merging**